### PR TITLE
Added additional test cases for the TimeStringParser.

### PR DESCRIPTION
**Pull request by RadtkeTomasz**
Following the convention of the file, I added some tests that fit more supported time units criteria.

Added an important CultureInfo.InvariantCulture argument to the Double parsing method (TimeStringParser line 72) which avoids the "String not in correct format" exception when testing from another country (the exception was thrown while performing original tests before any changes, in testing the "ExpectOk" "4.25 min" string).

### CODE REVIEW

**Comment by adamfk** [comment]
Thanks! Will check out after work tonight :)

File: src/StateSmithTest/Common/TimeStringParserTests.cs [hunk 1]
```diff
@@ -30,6 +44,12 @@ public void BadInput()
     {
         ExpectBadInput("+1.25 s");
         ExpectBadInput("5 long arse days");
+        ExpectBadInput("7 average looking horses");
+        ExpectBadInput("2 balls of ice cream");
+        ExpectBadInput("19 birthday cakes");
+        ExpectBadInput("0.9 successful relationships");
+        ExpectBadInput("100 days of coding");
+        ExpectBadInput("-15 blissful sniffs");
```
Comment (adamfk) [inline src/StateSmithTest/Common/TimeStringParserTests.cs:52]: Loved these! Made me laugh out loud 😆

**Review by adamfk — CHANGES_REQUESTED** [review]
Looks good! Just need to remove the .wsuo file I think, and remove the TODOs that you completed.

Thanks!

File: .vs/StateSmith/v17/.wsuo
[the remark this answers is not in the document]
Comment (RadtkeTomasz) [reply .vs/StateSmith/v17/.wsuo:1]: It was not - sorry. I wasn't aware of this file's creation - it was generated automatically. The .suo or .wsuo files are VS files to store local data about user solution options. I cannot even locate it through VS Solution Explorer to mark it as ignored, so I edited the gitignore file to include .wsuo. I hope this resolves the problem.

File: src/StateSmith/Common/TimeStringParser.cs [hunk 2]
```diff
@@ -68,7 +69,7 @@ private static void ParseNumberAndUnits(string elapsedTimeStr, out double number
         if (!match.Success)
             throw new ArgumentException("Failed to match elapsed time for input: " + elapsedTimeStr);
 
-        number = double.Parse(match.Groups["number"].Value);
+        number = double.Parse(match.Groups["number"].Value, CultureInfo.InvariantCulture);
```
[the remark this answers is not in the document]
Comment (RadtkeTomasz) [reply src/StateSmith/Common/TimeStringParser.cs:72]: I am glad I could contribute !

File: src/StateSmithTest/Common/TimeStringParserTests.cs [hunk 3]
```diff
@@ -13,15 +13,29 @@ public void ValidInput()
...
+        ExpectOk("0.65 min", (int)(0.65 * 60 * 1000));
+        ExpectOk("1 minute", (int)(1 * 60 * 1000));
+        ExpectOk("15 minutes", (int)(15 * 60 * 1000));
+
+        ExpectOk("52398 s", 52398 * 1000);
+        ExpectOk("777 secs", 777 * 1000);
+        ExpectOk("444 seconds", 444 * 1000);
+
+
         // TODO add all variations https://github.com/StateSmith/StateSmith/issues/131
```
[the remark this answers is not in the document]
Comment (RadtkeTomasz) [reply src/StateSmithTest/Common/TimeStringParserTests.cs]: Please check if it's okay now - I am not 100% sure that the wsuo file is now ignored or not. (I ignored the whole local .vs folder through the projects gitignore file). Thanks

**Review by adamfk — APPROVED** [review]

**Pull request merged**